examples/tv-casting-app/tv-casting-common/core/CastingPlayer.h

```diff
@@ -0,0 +1,145 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "Types.h"
+#include "lib/support/logging/CHIPLogging.h"
+
+#include <string.h>
+#include <vector>
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+
+namespace matter {
+namespace casting {
+namespace core {
+
+using namespace memory;
+//using namespace endpoint;
+
+
+enum ConnectionError
+{
+    NO_ERROR          = 0,
+    FAILED_TO_CONNECT = 1
+};
+
+using ConnectCallback    = std::function<void(ConnectionError)>;
+using DisconnectCallback = std::function<void(void)>;
+
+const int kPortMaxLength = 5; //port is uint16_t
+const int kIdMaxLength = chip::Dnssd::kHostNameMaxLength + kPortMaxLength;
+
+class Attributes
+{
+public:
+    char id[kIdMaxLength] = {};
+    char name[chip::Dnssd::kMaxDeviceNameLen + 1] = {};
+    char host_name[chip::Dnssd::kHostNameMaxLength + 1] = {};
+    size_t num_IPs; // number of valid IP addresses
+    chip::Inet::IPAddress ip_address[chip::Dnssd::CommonResolutionData::kMaxIPAddresses];
+    uint16_t product_id;
+    uint16_t vendor_id;
+    uint32_t type;
+};
+
+class CastingPlayer : public std::enable_shared_from_this<CastingPlayer>
+{
+private:
+    //std::vector<Strong<Endpoint>> endpoints;
+    bool connected = false;
+    Attributes attributes;
+
+public:
+    CastingPlayer(Attributes playerAttributes)
+    {
+        attributes = playerAttributes;
+    }
+    const char* GetId() const { return attributes.id; }
+
+    const char* GetName() const { return attributes.name; }
+
+    const char* GetHostName() const { return attributes.host_name; }
+
+    size_t GetNumIPs() const { return attributes.num_IPs; }
+
+    const chip::Inet::IPAddress& GetIPAddress(size_t idx) const { return attributes.ip_address[idx]; }
```

Comment:
This should be GetIPAddresses and we should return all the IPAddresses. The client can run through the indices on its own.

Reply:
updated